pinot-core/src/main/java/com/linkedin/pinot/core/segment/DefaultSegmentNameGenerator.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.segment;
+
+import com.linkedin.pinot.core.segment.creator.AbstractColumnStatisticsCollector;
+
+
+/**
+ * An interface that allows generates names for segments depending on the naming scheme.
+ */
+public interface DefaultSegmentNameGenerator {
+  String getSegmentName(AbstractColumnStatisticsCollector timeColStatsCollector, DefaultSegmentNameConfig segmentNameConfig) throws Exception;
```

Comment:
change signature to 
public String getSegmentName(AbstractColumnStatisticsColelctor timeColStatsCollector, SegmentNameConfig segmentNameConfig); 

Do we really want to throw "Exception"? If there are specific exceptions, then list them there. Do not throw high level Exception as far as possible